services/web/client/source/class/qxapp/desktop/PrjEditor.js

```diff
@@ -217,6 +214,30 @@ qx.Class.define("qxapp.desktop.PrjEditor", {
       }
     },
 
+    __addNode: function() {
+      if (this.__mainPanel.getMainView() !== this.__workbenchView) {
+        return;
+      }
+      this.__workbenchView.openServicesCatalogue();
+    },
+
+    __removeNode: function(nodeId) {
+      if (this.__mainPanel.getMainView() !== this.__workbenchView) {
+        return;
+      }
+      // remove first the connected links
+      let connectedLinks = this.getProjectModel().getWorkbenchModel().getConnectedLinks(nodeId);
```

Comment:
maybe worth saving this to a temporary variable ... `getProjectModel().getWorkbenchModel()`